dist/api_data.py

```diff
@@ -520,10 +520,13 @@ def __ge__(self, other):
         the maximum number of milliseconds an application thread will wait for space to be
         available in cache before giving up. Default will wait forever''',
         min=0),
-    Config('diagnostic_asserts', 'off', r'''
+    Config('diagnostic_asserts', '', r'''
         enable additional assertions.
         When \c off only mandatory assertions are enabled.
-        When \c on additional diagnostic assertions are enabled.''',
+        When \c on additional diagnostic assertions are enabled.
+        The default value is controlled by the compilation flag \c HAVE_DIAGNOSTIC. 
+        When \c HAVE_DIAGNOSTIC=1 this flag is \c on. 
+        When \c HAVE_DIAGNOSTIC=0 this flag is \c off''',
```

Comment:
The wording here is potentially a little confusing - we say the default is controlled by `HAVE_DIAGNOSTIC` then we say "When HAVE_DIAGNOSTIC=0 this flag is off". I think it would be clearer if the latter was "When HAVE_DIAGNOSTIC=0 this flag defaults to being off" (or something along those lines).

Reply:
Updated in [288006c](https://github.com/wiredtiger/wiredtiger/pull/8527/commits/288006c0484695136a579391b79673dca90983ce) 👍